### PR TITLE
Reduce the number of sql queries for articles comments #1606

### PR DESCRIPTION
**Pull request by lightalloy**
## What type of PR is this? (check all applicable)
- [x] Refactor

## Description
Eliminate N+1 while loading articles comments.

## Related Tickets & Documents
#1606 

## Added to documentation?
- [x] no documentation needed

### CODE REVIEW

File: app/view_objects/comments/tree.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+module Comments
```
Comment (rhymes) [inline app/view_objects/comments/tree.rb]: Love the introduction of view objects 💃

Comment (lightalloy) [reply app/view_objects/comments/tree.rb]: Thanks! Though I'm not sure if I'll keep it. I'm concerned about the memory usage cause of instantiating more objects (I'm checking it now)

Comment (rhymes) [reply app/view_objects/comments/tree.rb]: :-) in general there's a lot of logic in the views. It doesn't hurt to slowly make them more declarative anyhow

Comment (maestromac) [reply app/view_objects/comments/tree.rb]: Is this similar to the decorators we have in `/decorators` folder?

Comment (lightalloy) [reply app/view_objects/comments/tree.rb]: @maestromac they are similar in some terms. But they are also different: decorators are usually used to add functionality to an object, the draper ones that exist on a project add (often view-related) functionality to an instance of a model class. View objects are not tied to a specific object and contain view logic. They are often called presenters, but in my opinion this name is a bit confusing. I like the explanation of a view object from [a (classic) CodeClimate article](https://codeclimate.com/blog/7-ways-to-decompose-fat-activerecord-models/)

Comment (lightalloy) [reply app/view_objects/comments/tree.rb]: Probably I'll have to remove the view object from this pr, but I think this concept could be used for refactoring in other places.

Comment (maestromac) [reply app/view_objects/comments/tree.rb]: Thanks for the explanation. We definitely benefit from having some view objects

**Pull request merged**